### PR TITLE
[Remote-service] Add a path to save the model  @open sesame 11/17 16:24

### PR DESCRIPTION
**Pull request by gichan-jang**
Add a path to save the received model file to the ml-option.

Signed-off-by: gichan2-jang <gichan2.jang@samsung.com>

### CODE REVIEW

File: c/src/ml-api-service-remote.c [hunk 1]
```diff
@@ -371,13 +387,17 @@ _mlrs_process_remote_service (nns_edge_data_h data_h, void *user_data)
...
+      g_autofree gchar *dir_path = NULL;
 
       if (!_mlrs_get_data_from_uri ((gchar *) data, array)) {
         g_byte_array_free (array, TRUE);
         _ml_error_report_return (NNS_EDGE_ERROR_IO,
             "Failed to get data from uri: %s.", (gchar *) data);
       }
-      if (!_mlrs_model_register (service_key, data_h, array->data, array->len)) {
+
+      dir_path = _mlrs_get_model_dir_path (remote_s, service_key);
```
Comment (songgot) [inline c/src/ml-api-service-remote.c:398]: _mlrs_get_model_dir_path() may return null in an ERROR.
How about returning an error if dir_path is NULL?

File: c/src/ml-api-service-remote.c [hunk 2]
```diff
@@ -388,7 +408,9 @@ _mlrs_process_remote_service (nns_edge_data_h data_h, void *user_data)
     }
     case ML_REMOTE_SERVICE_TYPE_MODEL_RAW:
     {
-      if (!_mlrs_model_register (service_key, data_h, data, data_len)) {
+      g_autofree gchar *dir_path =
```
Comment (songgot) [inline c/src/ml-api-service-remote.c:411]: same

File: c/src/ml-api-service-remote.c (hunk 1 above)
Comment (gichan-jang) [reply c/src/ml-api-service-remote.c:398]: Thanks, updated!

File: c/src/ml-api-service-remote.c [hunk 3]
```diff
@@ -576,22 +610,26 @@ ml_service_remote_create (ml_option_h option, ml_service_event_cb cb,
   remote_s->node_type = edge_info->node_type;
   remote_s->event_cb = cb;
   remote_s->user_data = user_data;
+  if (ML_ERROR_NONE == ml_option_get (option, "path", &value)) {
```
Comment (anyj0527) [inline c/src/ml-api-service-remote.c:613]: may check `G_FILE_TEST_IS_DIR` here?

Comment (gichan-jang) [reply c/src/ml-api-service-remote.c:606]: Good, I changed to check `G_FILE_TEST_IS_DIR` here.

**Review by anyj0527 — APPROVED** [review]
LGTM✔

**Review by songgot — APPROVED** [review]
LGTM :+1:

File: c/src/ml-api-service-remote.c [hunk 4]
```diff
@@ -283,17 +273,36 @@ _mlrs_model_register (gchar * service_key, nns_edge_data_h data_h,
...
+    const gchar * service_key)
+{
+  g_autofree gchar *dir_path = NULL;
+
+  if (remote_s->path) {
+    dir_path = g_strdup (remote_s->path);
+  } else {
+    g_autofree gchar *current_dir = g_get_current_dir ();
+
+    dir_path = g_build_path ("/", current_dir, service_key, NULL);
```
Comment (jaeyun-jung) [inline c/src/ml-api-service-remote.c:295]: recommend using G_DIR_SEPARATOR_S instead '/'  :)

File: c/src/ml-api-service-remote.c [hunk 5]
```diff
@@ -283,17 +273,36 @@ _mlrs_model_register (gchar * service_key, nns_edge_data_h data_h,
     _ml_loge ("Failed to register model, service ket:%s", service_key);
     ret = FALSE;
```
Comment (jaeyun-jung) [inline c/src/ml-api-service-remote.c:274]: L284 and 294, need ret? IMO using autofree, ret can be removed.

File: c/src/ml-api-service-remote.c (hunk 4 above)
Comment (gichan-jang) [reply c/src/ml-api-service-remote.c]: Thanks, I changed them and other files.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**